lib/JITServer/JITServer.cpp

```diff
@@ -166,24 +166,119 @@ ServerInitializeThreadContext(
     *prereservedRegionAddr = 0;
 
     ServerThreadContext * contextInfo = nullptr;
-    try
+    HRESULT hr = S_OK;
+    HANDLE processHandle = (HANDLE)threadContextData->processHandle;
+    const DWORD stackArraySize = 1024;
+    HMODULE stackArray[stackArraySize];
+    DWORD neededBytes = 0;
+    HMODULE * moduleArray = stackArray;
+    DWORD arraySize = stackArraySize;
+
+    BEGIN_TRANSLATE_OOM_TO_HRESULT
     {
-        AUTO_NESTED_HANDLED_EXCEPTION_TYPE(static_cast<ExceptionType>(ExceptionType_OutOfMemory));
         contextInfo = (ServerThreadContext*)HeapNew(ServerThreadContext, threadContextData);
+
         ServerContextManager::RegisterThreadContext(contextInfo);
     }
-    catch (Js::OutOfMemoryException)
+    END_TRANSLATE_OOM_TO_HRESULT(hr);
+
+    if (FAILED(hr))
     {
-        return E_OUTOFMEMORY;
+        goto FailureCleanup;
+    }
+
+    // if our stack array does not have enough size, allocate an array large enough to handle all the modules
+    if (!EnumProcessModules(processHandle, stackArray, arraySize, &neededBytes))
+    {
+        Assert(neededBytes % sizeof(HMODULE) == 0);
+
+        // add 32 to give us reasonable buffer in case more modules are loading while we are querying
+        arraySize = neededBytes / sizeof(HMODULE) + 32;
```

Comment:
> + 32 [](start = 49, length = 5)

why not just use a while loop to query until succeed or non-size related error happened, instead of using this speculated size?